src/events/events.ts

```diff
@@ -106,6 +107,56 @@ const logTo = async (
     await logs.send({ embeds: [embed] });
 };
 
+/**
+ * Greets a new member or says goodbye to a member in the specified server.
+ * 
+ * @param member the member to greet or say goodbye to
+ * @param server the server document from the database
+ * @param action the action of welcoming or saying goodbye
+ */
+const greet = async (
+    client: Client,
+    member: GuildMember,
+    server: Server,
+    action: Action
+): Promise<void> => {
+    let greeting: Greeting;
+
+    switch (action) {
+        case Action.Welcome:
+            greeting = server.welcome;
+            break;
+        case Action.Goodbye:
+            greeting = server.goodbye;
+            break;
+    }
+
+    // No need to greet if there's no configuration or
+    // if the new member is a bot
+    if (!greeting || member.user.bot) {
+        return;
+    }
+
+    // channel is guaranteed to be a text channel due to the constraints
+    // of the /config welcome and /config goodbye command
+    const channel: TextChannel = <TextChannel> client.channels.cache.get(greeting.channel);
+
+    if (!channel) {
+        // Remove configurations if the channel does not exist
+        switch (action) {
+            case Action.Welcome:
+                server.welcome = null;
+                break;
+            case Action.Goodbye:
+                server.goodbye = null;
+                break;
+        }
+        await server.save();
+    }
+
+    await channel.send(await formatGreeting(member, action, greeting.message));
+};
```

Comment:
This is the helper function that welcomes new members says goodbye to members who leave. It stays silent when there aren't any configurations, or if the member in question is a bot.